src/codelabs/iterables/index.md

```diff
@@ -0,0 +1,571 @@
+---
+title: "Iterable Collections in Dart"
+description: Learn about and practice using Iterables in Dart!
+---
+
+This codelab teaches you how to use collections that implement the 
+[Iterable][iterable class] class, for example [List][list class] and
+[Set][set class]. Using the embedded DartPad editors, you can test your
+knowledge by running example code and completing exercises.
+
+To get the most out of this codelab, you should have the basic knowledge
+of the [Dart syntax](/samples).
+
+This codelab covers the following material:
+
+* How to access elements of an Iterable.
+* How to check if the elements of an Iterable match a condition.
+* How to filter the contents of an Iterable.
+* How to map the contents of an Iterable to a different value.
+
+Estimated time to complete this codelab: 60 minutes.
+
+{{site.alert.note}}
+  This page uses embedded DartPads to display examples and exercises.
+  {% include dartpads-embedded-troubleshooting.md %}
+{{site.alert.end}}
+
+## Why do you need collections
+
+Collections are classes that represented a group of objects, also called "elements".
+
+A collection can be empty, or contain many elements in it. Depending on its purpose,
+a collection will follow a different data structure and have different implementations.
+These are some of the most common collection types:
+
+* [List][list class]: Used to access elements by their index.
+* [Set][set class]: Used to contain elements that can only occur once.
+* [Map][map class]: Used to access elements using a key.
+
+In Dart, both `List` and `Set` are `Iterable`, meaning that they share the same methods as the 
+`Iterable` class. 
+
+## What is an Iterable?
+
+An `Iterable` is a collection of elements that can be accessed sequentially.
+
+`Iterable`, in Dart, is an abstract class, meaning that you cannot instantiate it directly.
+However you can create a new `Iterable` by creating a new `List` or a `Set`.
+ 
+This example shows a `List` of `int`, which is also an `Iterable` of `int`:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+{% endprettify %}
+
+The difference with a `List`, is that with the `Iterable` you cannot 
+access elements using their index.
+For example **the following code will give you an error**:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable[![1]!];
+{% endprettify %}
+
+If you access elements by the index, the compiler will tell you that
+`The operator '[]' isn't defined for the class 'Iterable'` which
+means that you cannot use `[index]` in this case.
+
+Continue to the next section to learn how to access elements of an `Iterable`.
+
+## Accessing elements
+
+The elements of an `Iterable` can be accessed sequentially.
+
+One way to do that, is by getting an `Iterator` using the `iterator` getter,
```

Comment:
The reason I added an example for it is because it is mentioned as well in the documentation of https://api.dartlang.org/stable/2.6.1/dart-core/Iterable-class.html before talking about the for-loop usage.

But I completely agree with you that it can be moved into a nice to know section.